ros2topic/ros2topic/verb/pub.py

```diff
@@ -115,6 +120,9 @@ def main(args):
     if args.once:
         times = 1
 
+    if args.stdin:
+        args.values = correct_stdin()
```

Comment:
```suggestion
    if args.stdin:
        values = collect_stdin()
    else:
        values = args.values
```

(and then change the argument passed to the function to use `values` instead of `args.values`)

Reply:
Fixed.